processing/src/main/java/org/apache/druid/query/rowsandcols/RearrangedRowsAndColumns.java

```diff
@@ -128,7 +130,16 @@ public boolean isNull(int rowNum)
             @Override
             public Object getObject(int rowNum)
             {
-              return accessor.getObject(pointers[start + rowNum]);
+              Object value = accessor.getObject(pointers[start + rowNum]);
+              if (ColumnType.STRING.equals(getType()) && value instanceof List) {
```

Comment:
Since we require this check for druid 31, we can live with this "hackery". Once https://github.com/apache/druid/pull/17038 gets merged, please remove this check. 
cc @Akshat-Jain